### PR TITLE
RSDK-2252 - Unable to run cartogropher through RDK

### PR DESCRIPTION
**Pull request by kkufieta**
Ticket: https://viam.atlassian.net/browse/RSDK-2252

slam side: https://github.com/viamrobotics/slam/pull/173

### CODE REVIEW

**Review by tessavitabile — APPROVED** [review]
LGTM mod comment

**Comment by zaporter-work** [comment]
This code will be changed and refactored in the future. Ticket: https://viam.atlassian.net/browse/RSDK-2267

File: services/slam/builtin/builtin.go [hunk 1]
```diff
@@ -535,7 +535,10 @@ func NewBuiltIn(ctx context.Context, deps registry.Dependencies, config config.S
 		return nil, errors.Wrap(err, "runtime slam config error")
 	}
 
-	svcConfig.SetOptionalParameters(localhost0, defaultDataRateMsec, defaultMapRateSec, logger)
+	port, dataRateMsec, mapRateSec, useLiveData, deleteProcessedData, err := slamConfig.SetOptionalParameters(svcConfig, localhost0, defaultDataRateMsec, defaultMapRateSec, logger)
```
[the remark this answers is not in the document]
Comment (nicksanford) [reply services/slam/builtin/builtin.go:538]: Yes this needs to be `SetOptionalParametersReplacement`

**Review by nicksanford — CHANGES_REQUESTED** [review]

**Review by zaporter-work — APPROVED** [review]

File: services/slam/builtin/builtin.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (kkufieta) [reply services/slam/builtin/builtin.go]: Done

**Review by nicksanford — APPROVED** [review]

**Review by nicksanford — APPROVED** [review]

**Review by nicksanford — APPROVED** [review]

**Pull request merged**